pilot/pkg/networking/core/cluster_tls.go

```diff
@@ -45,44 +45,50 @@ var istioMtlsTransportSocketMatch = &structpb.Struct{
 	},
 }
 
-var internalUpstreamSocket = &core.TransportSocket{
-	Name: "envoy.transport_sockets.internal_upstream",
-	ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: protoconv.MessageToAny(&internalupstream.InternalUpstreamTransport{
-		PassthroughMetadata: []*internalupstream.InternalUpstreamTransport_MetadataValueSource{
-			{
-				Kind: &metadata.MetadataKind{Kind: &metadata.MetadataKind_Host_{}},
-				Name: util.OriginalDstMetadataKey,
-			},
-			{
-				Kind: &metadata.MetadataKind{Kind: &metadata.MetadataKind_Cluster_{
-					Cluster: &metadata.MetadataKind_Cluster{},
-				}},
-				Name: "istio",
-			},
-			{
-				Kind: &metadata.MetadataKind{Kind: &metadata.MetadataKind_Host_{
-					Host: &metadata.MetadataKind_Host{},
-				}},
-				Name: "istio",
+func internalUpstreamSocket() *core.TransportSocket {
```

Comment:
why do you change this, is ther any mutating later

Reply:
Yes, the proxy protocol mutates it because it's wrapping the existing ones

Reply:
This is caught by cicd pretty easily. If you apply proxy protocol on any cluster with ambient it will crash pretty much immediately